graphql/course/mutations.ts

```diff
@@ -106,6 +109,30 @@ export class MutateCourseResolver {
         }
         const result = await prisma.course.update({ data, where: { id: courseId } });
         logger.info(`Course (${result.id}) updated by Student (${context.user.studentId})`);
+
+        const subcourse = await prisma.subcourse.findFirst({
+            where: { courseId: courseId },
+        });
+        const usersSubcourseAchievements = await prisma.user_achievement.findMany({
+            where: { context: { path: ['relation'], equals: `subcourse/${subcourse.id}` }, userId: user.userID },
+            include: { template: true },
+        });
+        const subcourseAchievements = await Promise.all(
+            usersSubcourseAchievements.map(async (usersSubcourseAchievement) => {
+                return await prisma.user_achievement.findMany({
+                    where: { context: { path: ['relation'], equals: `subcourse/${subcourse.id}` }, templateId: usersSubcourseAchievement.template.id },
+                });
+            })
+        );
```

Comment:
Isn't that the same as 
```
        const subcourses = await prisma.user_achievement.findMany({
            where: { context: { path: ['relation'], equals: `subcourse/${subcourse.id}` } },
            include: { template: true },
        });
```
?

Reply:
you are right, the userId is unnecessary here, having the id of a user_achievement